### DISCUSSION TITLE
How to select all rows wich ids in subquery?

### DISCUSSION THREAD

**Discussion by heckad** [sqlalchemy/sqlalchemy#7833] [Usage Questions] [answered]
I have a big query (for exmaple `select(SomeModel)`. I want to get all rows from another model where id in that query.
Simple code exmaple
```
some_query = select(SomeModel)

new_query = select(SomeModelLog).where(SomeModelLog.c.some_model_id.in_(some_query)
```
But I got an error `InvalidRequestError` with `Don't know how to join to .....` but I don't do any joins. I want to get something like
```
select * from some_model_log_table where id in (select some_model_id from some_model_table);
```
How to active that query?

**Comment by zzzeek** [comment, marked as answer]
this is essentially what [selectinload](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html?highlight=selectinload#sqlalchemy.orm.selectinload) does .

i think your query needs to select from just a single column:

```py
some_query = select(SomeModel.id)

new_query = select(SomeModelLog).where(SomeModelLog.c.some_model_id.in_(some_query)
```

I dont see anything about joins and that would be based on what your model looks like.

**Reply by heckad** [reply]
Yes, using `select(SomeModel.id)` helps. Big thanks.
